.github/workflows/codechecker.yaml

```diff
@@ -0,0 +1,62 @@
+name: Run Static Analysis
+
+on:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  RunCodechecker:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.9'
+
+      - name: Install Dependencies
+        run: |
+          curl -O https://efa-installer.amazonaws.com/aws-efa-installer-1.30.0.tar.gz
+          tar -xf aws-efa-installer-1.30.0.tar.gz && cd aws-efa-installer
+          sudo ./efa_installer.sh -y
+          # https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa-start-nccl-base.html#nccl-start-base-drivers
+          sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
+          sudo apt-key adv --fetch-keys http://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu2004/x86_64/7fa2af80.pub
+          wget -O /tmp/deeplearning.deb \
+              http://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu2004/x86_64/nvidia-machine-learning-repo-ubuntu2004_1.0.0-1_amd64.deb
+          sudo dpkg -i /tmp/deeplearning.deb
+          sudo wget -O /tmp/cuda.pin https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/cuda-ubuntu2004.pin
+          sudo mv /tmp/cuda.pin /etc/apt/preferences.d/cuda-repository-pin-600
+          sudo apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/3bf863cc.pub
+          sudo add-apt-repository 'deb http://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/ /'
```

Comment:
That said, you shouldn't need to grab these from the NVIDIA repo. If you are using an Ubuntu runner, just grab
https://github.com/aws/aws-ofi-nccl/blob/02dc354bab2c145f8cf4c91c6fb12ffb9f7df16d/.github/workflows/distcheck.yaml#L26

which is good enough to get a build going for the static analyzer.

Reply:
+1, thanks.

Reply:
Note that with some recent changes to the job on this PR, this happens way faster now.